include/ck_tile/ops/gemm/kernel/gemm_tile_partitioner.hpp

```diff
@@ -1,73 +1,130 @@
 // SPDX-License-Identifier: MIT
-// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
+// Copyright (c) 2018-2025, Advanced Micro Devices, Inc. All rights reserved.
 
 #pragma once
 
 #include "ck_tile/core.hpp"
 
 namespace ck_tile {
-template <typename BlockGemmShape_>
-struct GemmTilePartitioner
+
+/// @brief An internal ck-tile class dedicated to 2D tiles.
+template <typename BlockGemmShapeType>
+struct GemmTile2DPartitioner
 {
-    using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
+    using BlockGemmShape = remove_cvref_t<BlockGemmShapeType>;
 
-    static constexpr index_t kM = BlockGemmShape::kM;
-    static constexpr index_t kN = BlockGemmShape::kN;
-    static constexpr index_t kK = BlockGemmShape::kK;
+    static constexpr index_t MPerBlock = BlockGemmShape::kM;
+    static constexpr index_t NPerBlock = BlockGemmShape::kN;
+    static constexpr index_t KPerBlock = BlockGemmShape::kK;
 
-    CK_TILE_HOST static constexpr auto GridSize(index_t M, index_t N, index_t batch_size)
+    CK_TILE_HOST static constexpr auto GridSize(index_t M, index_t N, index_t batch_size) noexcept(
+        noexcept(MPerBlock != 0 && NPerBlock != 0)) -> dim3
     {
-        index_t GridDimX = (M + kM - 1) / kM;
-        index_t GridDimY = (N + kN - 1) / kN;
-        index_t GridDimZ = batch_size;
+        const index_t GridDimX = (M + MPerBlock - 1) / MPerBlock;
+        const index_t GridDimY = (N + NPerBlock - 1) / NPerBlock;
+        const index_t GridDimZ = batch_size;
         return dim3(GridDimX, GridDimY, GridDimZ);
     }
 
-    CK_TILE_HOST_DEVICE static constexpr auto GetLoopNum(index_t K)
+    CK_TILE_HOST_DEVICE static constexpr auto GetLoopNum(index_t K) noexcept -> index_t
     {
-        return integer_divide_ceil(K, kK);
+        return integer_divide_ceil(K, KPerBlock);
     }
 
-    CK_TILE_DEVICE auto operator()()
+    /// @brief Returns the output tile indexes. Call with blockIdx and blockIdy, where:
+    // blockIdx is blockId.x and blockIdy is blockIdx.y.
```

Comment:
Please refactor doc. Please use doxygen keywords for input arguments and output values.